circleci/templates/base_config.yml

```diff
@@ -0,0 +1,142 @@
+            fi
+            mvn test
+      - run:
+          name: Save test results
+          command: |
+            mkdir -p ~/test-results/junit/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/test-results/junit/ \;
+          when: always
+      - store_test_results:
+          path: ~/test-results
+      - store_artifacts:
+          path: ~/test-results/junit
+  owasp:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Owasp Tests
+          command: |
+            if [ $CIRCLE_PR_NUMBER ]
+            then
+              mvn install -B -Powasp -DskipTests -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            else
+              mvn install -B -Powasp -DskipTests -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            fi
+            cp **/dependency-check-report.* /tmp/artifacts
+      - store_artifacts:
+          path: /tmp/artifacts
+  sonar:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Sonar check
+          command: |
+            if [ $CIRCLE_BRANCH = master ]
+            then
+              if [ $RELEASE = true ]
+              then
+                git checkout $RELEASE_TAG
+              fi
+              mvn -q -B install sonar:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=cx -Dsonar.projectKey=$GITHUB_REPONAME -Dsonar.exclusions=$COVERAGE_EXCLUSIONS $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            fi
+  release_tag:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Release Tag
+          command: |
+            if [ $RELEASE = true ]
+            then
+              git checkout $RELEASE_COMMIT
+              git remote add ssh-origin git@github.com:connexta/$CIRCLE_PROJECT_REPONAME.git
+              echo ssh -i $GITHUB_KEY -l git -o StrictHostKeyChecking=no \\"\\$@\\" > run_ssh.sh
+              chmod +x run_ssh.sh
+              git push ssh-origin HEAD:$CIRCLE_BRANCH && git push ssh-origin $RELEASE_TAG
+            fi
+  deploy:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Deploy
+          command: |
+            if [ $RELEASE = true ]
+            then
+              git checkout $RELEASE_TAG
+            fi
+            mvn javadoc:aggregate -B -DskipTests -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            if [ $RELEASE = true]
+            then
+              mvn deploy -B -DskipTests -DretryFailedDeploymentCount=10 -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS -Dgpg.secretKeyring=$ION_GPG_KEYRING -Dgpg.publicKeyring=$ION_GPG_KEYRING -Dgpg.passphraseServerId=\"ion-signing\"
+            else
+              mvn deploy -B -DskipTests -DretryFailedDeploymentCount=10 -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            fi
+
+workflows:
+  version: 2.1
+  nightly:
```

Comment:
This isn't really a nightly workflow as there is no cron timer here. This workflow will run by default with every commit to the branch. I think this should be broken up into two workflows.

Workflow A: build, test, owasp, sonar (this is for PR branches)
Workflow B: build, test, owasp, sonar, tag release, deploy (this is for commits to master)

This can work by applying a filter to the workflow to only operate for master/non-master branches.